junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -120,8 +272,13 @@ inline fun <reified T : Throwable> assertThrows(executable: () -> Unit): T {
  * ```
  * @see Assertions.assertThrows
  */
-inline fun <reified T : Throwable> assertThrows(message: String, executable: () -> Unit): T =
-    assertThrows({ message }, executable)
+inline fun <reified T : Throwable> assertThrows(message: String, executable: () -> Unit): T {
+    contract {
+        callsInPlace(executable)
+    }
+
+    return assertThrows({ message }, executable)
```

Comment:
Do you know why this is not calling the Java method directly? Why is the overload so tricky with exceptions?

Reply:
I assume it's done this way to support suspending functions.
[Related issue](https://github.com/junit-team/junit5/issues/1851)